Fix issues 122 and 123

sangeetabhatia03 opened the pull request:
This PR fixes issues 122 and 123 highlighted by Pierre, and edits to code suggested by Anne.
Have run the full suite of tests and all tests pass.

sangeetabhatia03 committed:
Use correct epsilon_init while drawing R

All tests pass OK.

sangeetabhatia03 committed:
Test for more than 2 variants needs to be fixed

I think it is set-up incorrectly.

sangeetabhatia03 committed:
Pierre's suggestion for Issue 123

sangeetabhatia03 committed:
Use seq instead of colon syntax

annecori commented on tests/testthat/test-samplers.R:
```diff
@@ -307,6 +307,8 @@ test_that("estimate_joint produces expected results (>2 variants 4 locs)", {
   x <- estimate_joint(incid, si_distr, priors, seed = 1)
 
   ## epsilon should be approximately 1
+  ## FIXME this should be apply(x$epsilon, 1, mean)
```
@sangeetabhatia03  any chance you can fix this please then?

sangeetabhatia03 committed:
Fix all tests with more than 2 variants 

added a new test with different transmission advantages for the 2 variants, doesn't work for me atm.

sangeetabhatia03 committed:
Modified all tests with more than 2 variants

sangeetabhatia03 commented:
@annecori I have now modified all tests that had more than 2 variants.
